### PR TITLE
Update to Microphysics for HRRRv5

### PR DESCRIPTION
**Pull request by AndersJensen-NOAA**
This PR is built on top of: https://github.com/NOAA-GSL/MPAS-Model/pull/9

This PR includes:
1. Addition of a 3D reflectivity diagnostic for the microphysics.
2. Move the call to the reflectivity calculation from the MPAS microphysics driver to the microphysics scheme.
3. Namelist configuration flags for aerosol aware and hail aware microphysics.
4. Changes to the microphysics driver so that aerosol aware and non aerosol aware microphysics can be call (hail aware microphysics is NOT hooked up and has NOT been tested).
6. Added code framework to allow for the initialization of cloud, rain, and ice number concentrations (NOT hooked up in the microphysics submodule and NOT yet tested).
7. Added flags to the lookup table functionality. 

3-km test runs successfully.

### CODE REVIEW

File: src/core_atmosphere/physics/physics_wrf/module_sf_ruclsm.F [hunk 1]
```diff
@@ -574,10 +574,6 @@ subroutine lsmruc(spp_lsm,                                   &
          enddo
      enddo
 
-        do k=1,nsl
-           soilice(k)=0.
-           soiliqw(k)=0.
-        enddo
```
Comment (tanyasmirnova) [inline src/core_atmosphere/physics/physics_wrf/module_sf_ruclsm.F]: @jderrico-noaa Janet, could we revert removing lines 577-580? After all I think it is safer to have this initialization to zero.

**Review by joeolson42 — APPROVED** [review]

**Comment by joeolson42** [comment]
Hey everyone, let's get this reviewed asap!

**Review by mdtoyNOAA — APPROVED** [review]

File: src/core_atmosphere/physics/physics_wrf/module_sf_ruclsm.F [hunk 2]
```diff
@@ -574,10 +574,6 @@ subroutine lsmruc(spp_lsm,                                   &
          enddo
      enddo
```
Comment (jderrico-noaa) [inline src/core_atmosphere/physics/physics_wrf/module_sf_ruclsm.F]:
```suggestion

        do k=1,nsl
           soilice(k)=0.
           soiliqw(k)=0.
        enddo
```

Comment (tanyasmirnova) [reply src/core_atmosphere/physics/physics_wrf/module_sf_ruclsm.F]: @jderrico-noaa Thank you, Janet.

**Review by tanyasmirnova — APPROVED** [review]

**Pull request merged**